src/libsync/abstractcorejob.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+
+#include "owncloudlib.h"
+
+namespace OCC {
+
+class OWNCLOUDSYNC_EXPORT Job : public QObject
+{
+    Q_OBJECT
+
+    friend class AbstractCoreJobFactory;
+
+public:
+    explicit Job(QObject *parent = nullptr);
+
+    [[nodiscard]] const QVariant &result() const;
+
+    [[nodiscard]] const QString &errorMessage() const;
+    [[nodiscard]] QNetworkReply::NetworkError networkError() const;
+
+    [[nodiscard]] bool success() const;
+
+protected:
+    void finishWithResult(const QVariant &result);
```

Comment:
The name looks like a signal, maybe setResult?

Reply:
It does trigger the "this job has finished" signal, though. `set*` doesn't transport that message.

We should reconsider removing the `friend` class stuff and using public setters.